core/services/ocr2/plugins/ocr2keeper/evm/offchain_lookup.go

```diff
@@ -0,0 +1,349 @@
+package evm
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"math/big"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/itchyny/gojq"
+	"github.com/patrickmn/go-cache"
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/ocr2keepers/pkg/types"
+
+	"github.com/smartcontractkit/chainlink/core/gethwrappers/generated/keeper_registry_wrapper2_0"
+)
+
+const ParseFieldError = "field error"
+
+type OffchainLookup struct {
+	url              string
+	extraData        []byte
+	fields           []string
+	callbackFunction [4]byte
+}
+
+type OffchainLookupBody struct {
+	sender string
+	data   string
+}
+
+// offchainLookup looks through check upkeep results looking for any that need off chain lookup
+func (r *EvmRegistry) offchainLookup(ctx context.Context, upkeepResults []types.UpkeepResult) ([]types.UpkeepResult, error) {
+	for i, _ := range upkeepResults {
+		block, upkeepId, err := blockAndIdFromKey(upkeepResults[i].Key)
+		if err != nil {
+			r.lggr.Error("[OffchainLookup] error getting block and upkeep id:", err)
+			continue
+		}
+		// if its another reason continue/skip
+		if upkeepResults[i].FailureReason != UPKEEP_FAILURE_REASON_TARGET_CHECK_REVERTED {
+			continue
+		}
+
+		// checking if this upkeep is in cooldown from api errors
+		_, onIce := r.cooldownCache.Get(upkeepId.String())
+		if onIce {
+			r.lggr.Infof("[OffchainLookup] cooldown Skipping UpkeepId: %s\n", upkeepId)
+			continue
+		}
+
+		var offchainLookup OffchainLookup
```

Comment:
I think you might be able to drop this line, and then change the following line to

```
offchainLookup, err := r.decodeOffchainLookup(upkeepResults[i].PerformData)
```